.github/workflows/image-multiarch.yaml

```diff
@@ -0,0 +1,67 @@
+name: Build & push multi-arch image
+
+on:
+  workflow_call:
+    secrets:
+      GH_TOKEN:
+        description: "GitHub token"
+        required: true
+      ECR_REGISTRY:
+        description: "Repository to push the image"
+        required: true
+    inputs:
+      service:
+        description: "Name to tag the image"
+        type: string
+        required: true
+      build-args:
+        description: "Build args"
+        type: "string"
+        required: false
+      context:
+        description: "Build context"
+        type: string
+        required: false
+        default: "."
+      labels:
+        description: "Image labels"
+        type: string
+        required: false
+jobs:
+  build:
+    name: Build and push image
+    runs-on: [ self-hosted, bear-ephemeral ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker context for Buildx
+        run: docker context create builders
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          endpoint: builders
+      - name: Build and push images
+        uses: docker/build-push-action@v5
+        with:
+          build-args: ${{ inputs.build-args }}
+          context: ${{ inputs.context }}
+          labels: ${{ inputs.labels }}
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: ${{ secrets.ECR_REGISTRY }}/${{ inputs.service }}:${{ github.run_id }}
```

Comment:
Let's remember that we also have a "standard" to push images with `-dev` prefix in the case of "test" images.

Reply:
And we also have serverless versioning which also has func name there, so I will require version as input for now unless we will have a proper convention for image tagging